x-pack/plugins/lens/public/indexpattern_datasource/datapanel.tsx

```diff
@@ -510,32 +545,43 @@ export const InnerIndexPatternDataPanel = function InnerIndexPatternDataPanel({
                 }
               >
                 <EuiSpacer size="s" />
-                {!!filteredFieldGroups.availableFields.length ? (
-                  <div className="lnsInnerIndexPatternDataPanel__fieldItems">
-                    {paginatedAvailableFields.map((field: IndexPatternField) => {
-                      return (
-                        <FieldItem
-                          core={core}
-                          data={data}
-                          indexPattern={currentIndexPattern}
-                          key={field.name}
-                          field={field}
-                          highlight={hilight}
-                          exists={true}
-                          dateRange={dateRange}
-                          query={query}
-                          filters={filters}
-                        />
-                      );
-                    })}
-                  </div>
+                {hasSyncedExistingFields ? (
+                  !!filteredFieldGroups.availableFields.length ? (
+                    <div className="lnsInnerIndexPatternDataPanel__fieldItems">
+                      {paginatedAvailableFields.map((field: IndexPatternField) => {
+                        return (
+                          <FieldItem
+                            core={core}
+                            data={data}
+                            indexPattern={currentIndexPattern}
+                            key={field.name}
+                            field={field}
+                            highlight={hilight}
+                            exists={true}
+                            dateRange={dateRange}
+                            query={query}
+                            filters={filters}
+                          />
+                        );
+                      })}
+                    </div>
+                  ) : (
+                    <NoFieldsCallout
+                      isAffectedByGlobalFilter={!!filters.length}
+                      isAffectedByFieldFilter={hasFieldFilter}
+                      isAffectedByTimerange={true}
+                      existFieldsInIndex={!!fieldGroups.emptyFields.length}
+                    />
+                  )
                 ) : (
-                  <NoFieldsCallout
-                    isAffectedByGlobalFilter={!!filters.length}
-                    isAffectedByFieldFilter={hasFieldFilter}
-                    isAffectedByTimerange={true}
-                    existFieldsInIndex={!!fieldGroups.emptyFields.length}
-                  />
+                  <>
+                    <EuiSpacer size="m" />
+                    <EuiFlexGroup justifyContent="spaceAround">
+                      <EuiFlexItem grow={false}>
+                        <EuiLoadingSpinner size="m" />
```

Comment:
I like the loading spinner, but we already had a loading indicator on the entire panel for this. This now creates two indicators for the same thing. I'm okay with it, but I do think @cchaos should opine.
 
<img width="299" alt="Screenshot 2020-06-18 14 56 41" src="https://user-images.githubusercontent.com/666475/85061395-be954e00-b174-11ea-9150-23c1a60dfa5a.png">

Reply:
I completely agree - that's why I named the comment 'controversial spinner' :D
I was trying to find a solution for the problem we currently have - glitch of NoFieldsCallout before existing fields are loaded, but I wanted to talk about it during sync. Video from the 7.8 cloud version below:
https://www.loom.com/share/3c398f5296be4cf3a89755041300e8d8

Reply:
Instead of replacing the accordion content with the spinner, can you replace the notification count instead? This will better indicate that the number of fields is being updated as well.